pkg/config/interface.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"sigs.k8s.io/kubebuilder/v3/pkg/model/resource"
+)
+
+// Config defines the interface that project configuration types must follow
+type Config interface {
+	/* Version */
+
+	// GetVersion returns the current project version
+	GetVersion() Version
+
+	/* String fields */
+
+	// GetDomain returns the project domain
+	GetDomain() string
+	// SetDomain sets the project domain
+	SetDomain(domain string) error
+
+	// GetRepository returns the project repository.
+	GetRepository() string
+	// SetRepository sets the project repository
+	SetRepository(repository string) error
+
+	// GetName returns the project name
+	// This method was introduced in project version 3-alpha.
+	GetName() string
```

Comment:
```suggestion
	GetProjectName() string
```


See config.getName shows the name of the config when config.GetProjectName makes clear whet exactly is it.

Reply:
`Config` is the project configuration, so I think that the "project" part is implicit.

Let's have a third opinion, can you PTAL @estroz

Reply:
`GetProjectName()` is fine with me.